### PR TITLE
Document single user-sync endpoint

### LINKED ISSUE #2105: Document Single User Sync
Docs needed for
https://github.com/owncloud/core/pull/36428

## What Type Of Content Change Is This? (Optional)
- [x] New Content Addition
- [ ] Old Content Deprecation
- [ ] Existing Content Simplification
- [ ] Bug Fix to Existing Content

## Which Manual Does This Relate To? (Optional)
- [x] Admin Manual
- [ ] Developer Manual
- [ ] User Manual
- [ ] Android
- [ ] iOS
- [ ] Branded Clients
- [ ] Desktop Client
- [ ] Other

### PR DESCRIPTION
**Pull request by settermjd**
This fixes #2105.

### CODE REVIEW

File: modules/developer_manual/pages/core/apis/ocs/user-sync-api.adoc [hunk 1]
```diff
@@ -0,0 +1,93 @@
+= OCS User Sync API
+:toc: right
+:toclevels: 1
+:request-base-path: ocs/v2.php
+
+This endpoint triggers user-sync for a specific user.
```
Comment (mmattel) [inline modules/developer_manual/pages/core/apis/ocs/user-sync-api.adoc]: Would be great if you could add a `== Introduction` section as we have in all other documents 😃

Comment (settermjd) [reply modules/developer_manual/pages/core/apis/ocs/user-sync-api.adoc]: Sure. Thanks for spotting that.

Comment (settermjd) [reply modules/developer_manual/pages/core/apis/ocs/user-sync-api.adoc]: Done.

File: modules/developer_manual/pages/core/apis/ocs/user-sync-api.adoc [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+|`userid` 
+|string 
+|The id of the user to trigger a sync for.
+|===
+
+== Returns
+
+The request returns an:
+
+* `HTTP 200 OK` - If the user sync was executed.
```
Comment (mmattel) [inline modules/developer_manual/pages/core/apis/ocs/user-sync-api.adoc]: Have not seen it earlier, but it woud be easier to read if the results are put into a table

Comment (settermjd) [reply modules/developer_manual/pages/core/apis/ocs/user-sync-api.adoc]: Sure. Fair enough.

**Comment by settermjd** [comment]
@mmattel, the requested changes have been implemented.

**Comment by pmaier1** [comment]
This should be documented in the admin docs as well. Not necessary the whole thing, maybe just a reference to this way of triggering user sync. Somewhere around here https://doc.owncloud.com/server/admin_manual/configuration/user/user_auth_ldap.html#syncing-users.

**Comment by settermjd** [comment]
> This should be documented in the admin docs as well. Not necessary the whole thing, maybe just a reference to this way of triggering user sync. Somewhere around here https://doc.owncloud.com/server/admin_manual/configuration/user/user_auth_ldap.html#syncing-users.

Shall do that now.

**Comment by settermjd** [comment]
@pmaier1?

File: modules/admin_manual/pages/configuration/user/user_auth_ldap.adoc [hunk 3]
```diff
@@ -644,7 +644,7 @@ Whenever users log in, their display name, email, quota, avatar and search attri
 If you want to keep the metadata up to date you can set up a cron job, using xref:configuration/server/occ_command.adoc#syncing-user-accounts[the occ user:sync command].
 Versions of ownCloud before 10.0 imported all users when the users page was loaded, but this is no longer the case.
 
-TIP: We recommend xref:configuration/server/background_jobs_configuration.adoc#cron-jobs[creating a Cron job], to automate regularly syncing LDAP users with your ownCloud database.
+TIP: We recommend xref:configuration/server/background_jobs_configuration.adoc#cron-jobs[creating a Cron job], to automate regularly syncing LDAP users with your ownCloud database. You can use the xref:developer_manual:core/apis/ocs/user-sync-api.adoc[the OCS User Sync API] to simplify the process.
```
Comment (pmaier1) [inline modules/admin_manual/pages/configuration/user/user_auth_ldap.adoc]: - You have double "the"
- I don't think that the user sync API replaces or simplifies the cron job. It rather provides a way to trigger user sync from the outside which helps as the external user provisioning system can tell ownCloud when there are changes (push vs. pull). To put it differently: If you have many users you do not have to sync all of them (pull) to get one new user. Instead the external user backend "reports" that one new user via the API (push).
- Usually, you would still use the cron job to sync users.

Comment (settermjd) [reply modules/admin_manual/pages/configuration/user/user_auth_ldap.adoc]: I updated the section based on your feedback, @pmaier1.

**Comment by settermjd** [comment]
> THX +1

yw.

**Pull request merged**